app/models/document.rb

```diff
@@ -230,10 +230,7 @@ def publish
 
   def unpublish
     handle_remote_error do
-      Services.publishing_api.unpublish(content_id, type: 'gone')
-
-      AttachmentDeleteWorker.perform_async(content_id)
-      RummagerDeleteWorker.perform_async(base_path)
+      DocumentUnpublisher.unpublish(content_id, base_path)
```

Comment:
That's my thinking too. Giving the method a descriptive name forces more thought about the behaviour. It also communicates intent in places where the receiver isn't known (`expect(described_class.call).to eql(...)`).